spring-cloud-azure-appconfiguration-config/src/main/java/com/microsoft/azure/spring/cloud/config/stores/ClientStore.java

```diff
@@ -72,38 +81,51 @@ private ConfigurationAsyncClient buildClient(String store) throws IllegalArgumen
         if (tokenCredential != null) {
             // User Provided Token Credential
+            LOGGER.debug("Connecting to " + endpoint + " using AppConfigurationCredentialProvider.");
             builder.credential(tokenCredential);
         } else if ((connection.getClientId() != null && StringUtils.isNotEmpty(connection.getClientId()))
                 && connection.getEndpoint() != null) {
             // User Assigned Identity - Client ID through configuration file.
+            LOGGER.debug("Connecting to " + endpoint + " using Client ID from configuration file.");
             ManagedIdentityCredentialBuilder micBuilder = new ManagedIdentityCredentialBuilder()
                     .clientId(connection.getClientId());
             builder.credential(micBuilder.build());
         } else if (StringUtils.isNotEmpty(connection.getConnectionString())) {
             // Connection String
+            LOGGER.debug("Connecting to " + endpoint + " using Connecting String.");
             builder.connectionString(connection.getConnectionString());
         } else if (connection.getEndpoint() != null) {
-            // System Assigned Identity. Needs to be checked last as all of the above should have a Endpoint.
+            // System Assigned Identity. Needs to be checked last as all of the above
+            // should have a Endpoint.
+            LOGGER.debug("Connecting to " + endpoint
+                    + " using Azure System Assigned Identity or Azure User Assigned Identity.");
             ManagedIdentityCredentialBuilder micBuilder = new ManagedIdentityCredentialBuilder();
             builder.credential(micBuilder.build());
         } else {
             throw new IllegalArgumentException("No Configuration method was set for connecting to App Configuration");
         }
-        return builder.endpoint(endpoint).buildAsyncClient();
+
+        builder.endpoint(endpoint);
+
+        if (clientProvider != null) {
+            clientProvider.setup(builder, endpoint);
+        }
+
+        return builder.buildAsyncClient();
     }
 
     /**
-     * Gets a list of Configuration Settings from the revisions given config store that
+     * Gets the latest Configuration Setting from the revisions given config store that
      * match the Setting Selector criteria.
      * 
      * @param settingSelector Information on which setting to pull. i.e. number of
      * results, key value...
      * @param storeName Name of the App Configuration store to query against.
      * @return List of Configuration Settings.
      */
-    public final List<ConfigurationSetting> listSettingRevisons(SettingSelector settingSelector, String storeName) {
+    public final ConfigurationSetting getRevison(SettingSelector settingSelector, String storeName) {
         ConfigurationAsyncClient client = buildClient(storeName);
-        return client.listRevisions(settingSelector).collectList().block();
+        return client.listRevisions(settingSelector).blockFirst();
```

Comment:
Sorry for ask, but do we have how to know if the first in this list is the latest configuration from the revisions?